### DISCUSSION TITLE
"@storybook/preview-web"' has no exported member 'WebProjectAnnotations'.

### DISCUSSION THREAD

**Discussion by hamzalatif-e2x** [storybookjs/storybook#18531] [Help] [answered]
The latest version (6.5.9) of @storybook/preview-web does not have the WebProjectAnnotations type exported. @storybook/preview-web is a peer dependency of @storybook/testing-react which is using version >= 6.4.0 which results in version 6.5.9 being used with the missing export.

Version 6.4.22 of @storybook/preview-web had the export in https://github.com/storybookjs/storybook/blob/v6.4.22/lib/preview-web/src/types.ts

```
../../node_modules/@storybook/testing-react/dist/types.d.ts:2:15 - error TS2305: Module '"@storybook/preview-web"' has no exported member 'WebProjectAnnotations'.

import type { WebProjectAnnotations } from '@storybook/preview-web';
                ~~~~~~~~~~~~~~~~~~~~~

Found 1 error in ../../node_modules/@storybook/testing-react/dist/types.d.ts:2
```

Is there a reason for this omission or is this a bug?

**Comment by AhmadKabakibi** [comment]
Hi! 👋 

I used [patch-package](https://github.com/ds300/patch-package) to patch `@storybook/preview-web@6.5.9` for the project I'm working on.
I'm not sure if exporting WebProjectAnnotations from index.ts is the right solution but here is the diff that solved my problem:

```diff
diff --git a/node_modules/@storybook/preview-web/dist/ts3.9/index.d.ts b/node_modules/@storybook/preview-web/dist/ts3.9/index.d.ts
index f25c732..3477211 100644
--- a/node_modules/@storybook/preview-web/dist/ts3.9/index.d.ts
+++ b/node_modules/@storybook/preview-web/dist/ts3.9/index.d.ts
@@ -1,4 +1,4 @@
-export { composeConfigs } from '@storybook/store';
+export { composeConfigs, WebProjectAnnotations } from '@storybook/store';
 export { Preview } from './Preview';
 export { PreviewWeb } from './PreviewWeb';
 export { simulatePageLoad, simulateDOMContentLoaded } from './simulate-pageload';
```

**Comment by ingokpp** [comment]
Is there any reliable solution now?

**Comment by clintandrewhall** [comment]
An issue has been opened, but `testing-react` is still pinned to 6.4.x, so until they update its dependencies and release, I don't see how we can fix it.

https://github.com/storybookjs/testing-react/issues/106

**Reply by RiteshMakan** [reply]
For now, this solution seems to be working for me:
https://github.com/storybookjs/storybook/discussions/18531#discussioncomment-3098813

**Comment by TadeuA** [comment]
Hello guys, finally i've founded the solution for this problem, kindly, could you review this pr for me ? i really aprecciate that. Thanks


https://github.com/storybookjs/testing-react/pull/116

**Comment by clintandrewhall** [comment, marked as answer]
I forgot to mention here: I worked with the Storybook team and a PR has been merged for a `patch` release.

https://github.com/storybookjs/storybook/pull/19048
